src/main/java/com/fasterxml/jackson/dataformat/xml/util/DefaultXmlPrettyPrinter.java

```diff
@@ -501,35 +529,25 @@ public void writeIndentation(JsonGenerator g, int level) throws IOException
      * Default linefeed-based indenter uses system-specific linefeeds and
      * 2 spaces for indentation per level.
      */
-    protected static class Lf2SpacesIndenter
+    protected class Lf2SpacesIndenter
         implements Indenter, java.io.Serializable
     {
         private static final long serialVersionUID = 1L;
 
-        final static String SYSTEM_LINE_SEPARATOR;
-        static {
-            String lf = null;
-            try {
-                lf = System.getProperty("line.separator");
-            } catch (Throwable t) { } // access exception?
-            SYSTEM_LINE_SEPARATOR = (lf == null) ? "\n" : lf;
-        }
-
         final static int SPACE_COUNT = 64;
-        final static char[] SPACES = new char[SPACE_COUNT];
-        static {
+        final char[] SPACES = new char[SPACE_COUNT];
```

Comment:
Why would this be allocated separately for each instance? Let's keep `static`

Reply:
> Or are non-static inner classes not allowed to have static members? (that'd be the reason)

Yeah, java does not allow non-static inner classes having static members.
But can we move `char[] SPACES`  to outer `DefaultXmlPrettyPrinter` class?

Reply:
Yes, moving it there probably makes sense to avoid allocating these instances.

Reply:
> Yes, moving it there probably makes sense to avoid allocating these instances.

DONE! Moved `final static char[] SPACES` to DefaultXmlPrettyPrinter.